### ISSUE TITLE
Сортировка метаданных расширения по базовому проекту

### ISSUE THREAD

**Issue by DoublesunRUS** [1C-Company/v8-code-style#44]
Хочется иметь возможность для расширения включить автосортировку заимствованных метаданных так, как они отсортированы в базовом проекте.

**Comment by marmyshev** [comment]
А при этом базовый проект сортировать не хочется? и что делать с добавленными метаданными - их не сортировать?

И главное зачем это? Это пожелание не соответствует стандарту, поэтому нужно понимать "зачем это надо"...

**Comment by DoublesunRUS** [comment]
Базовый проект сортируют разработчики типовой конфигурации. Они реквизиты не сортируют.
Добавленные метаданные помещать в конце типовых.
Сортировать их или нет, зависит от настроек.
То есть я вижу себе некую галочку в дополнении к остальным настройкам "Сортировать заимствованные объекты аналогично базовым".

**Comment by marmyshev** [comment]
Я правильно понимаю, что ты хочешь чтобы в расширении:
1. Сначала шли заимствованные объекты и они должны быть отсортированы идентично сортировке конфигурации (даже если там она не правильная)
2. Потом отдельным списком должны идти добавленные объекты и они должны быть отсортированы по алфавиту

Верно?

**Comment by DoublesunRUS** [comment]
Да, верно. Причем п1 и п2 это две галочки. Соответственно каждый пункт выполняется только если включена соответствующая галочка.

**Comment by marmyshev** [comment]
Автосортировка и стандарт - они сделаны не для удобства разработка - а для удобства потребителя. Т.е. нужно делать как им будет удобно смотреть. 

Стандарт как раз препятствует различным схемам сортировки которые придумывают разработчики чтобы им было удобно.

Есть ли причины по которым такая хаотичная сортировка будет полезна потребителям?

**Comment by DoublesunRUS** [comment]
Мне сложно прокомментировать фразу "хаотичная сортировка". Если разработчики типовой выбрали вот такой порядок реквизитов, кто я такой чтобы им указывать что так делать неправильно?

**Comment by marmyshev** [comment]
Давай не перескакивать с пункта на пункт.
1. есть сортировка верхнего уровня метаданных - она обязательна
2. Есть сортировка реквизитов, форм и прочего - она не обязательно (в некоторых случаях запрещена, например порядок измерений)

Если проект поставщика не сортирует по п.2 (реквизиты) - что ты хочешь делать в расширении? хочешь сортировать сортируй - но смысл их сортировать? Если он есть - наверное он должен быть для потребителям полезен?

А по поводу п.1 - если проект поставщика отсортирован - ты хочешь хаотичную сортировку для потребителя всё равно? Можешь привести аргументы - какая польза потребителям от такой сортировки?

**Comment by DoublesunRUS** [comment]
Если проект поставщика не сортирует реквизиты, я хочу чтобы в расширении их порядок был таким же, каков он в базовом проекте.

**Comment by NikitaMikhaylovSB** [comment]
Погодите, сортировать просто так - нельзя, это же вызовет реструктуризацию...
